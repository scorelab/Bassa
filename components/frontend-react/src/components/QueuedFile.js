import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

// MUI imports
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Fab from '@material-ui/core/Fab';
import DeleteIcon from '@material-ui/icons/DeleteOutlined';
import red from '@material-ui/core/colors/red';

const CustomTableCell = withStyles(theme => ({
  head: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white
  },
  body: {
    fontSize: 14
  }
}))(TableCell);

const styles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
    overflowX: 'auto'
  },
  row: {
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.background.default
    }
  },
  icon: {
    marginRight: theme.spacing(2),
    fontSize: 20
  },
  fab: {
    color: theme.palette.getContrastText(red[500]),
    backgroundColor: red[500],
    '&:hover': {
      backgroundColor: red[700]
    }
  }
}));

const QueuedFile = props => {
  const classes = styles();
  const { index, name, onDelete } = props;
  return (
    <div className={classes.root}>
      <Table>
        <TableBody>
          <TableRow>
            <CustomTableCell>{index + 1}</CustomTableCell>
            <CustomTableCell className={classes.row} component="th" scope="row">
              {name}
            </CustomTableCell>
            <CustomTableCell align="right">
              <Fab
                data-test="button-delete"
                size="small"
                aria-label="Delete"
                className={classes.fab}
                onClick={onDelete}
              >
                <DeleteIcon />
              </Fab>
            </CustomTableCell>
          </TableRow>
        </TableBody>
      </Table>
    </div>
  );
};

QueuedFile.propTypes = {
  name: PropTypes.string.isRequired
};

export default QueuedFile;
