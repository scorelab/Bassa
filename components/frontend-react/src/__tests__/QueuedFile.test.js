/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import QueuedFile from '../components/QueuedFile';

const clickDeleteFn = jest.fn();

const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

describe('Single queued file component', () => {
  let component;

  beforeEach(
    // eslint-disable-next-line no-return-assign
    () =>
      (component = shallow(
        <QueuedFile name="Name of the file" onDelete={clickDeleteFn()} />
      ))
  );

  it('should match with its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call onDelete function on button click', () => {
    const wrapper = findByTestAttr(component, 'button-delete');
    wrapper.simulate('click');
    expect(clickDeleteFn).toHaveBeenCalled();
  });
});
