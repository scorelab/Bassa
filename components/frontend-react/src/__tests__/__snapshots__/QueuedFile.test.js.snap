// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single queued file component should match with its snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(WithStyles(ForwardRef(TableCell)))>
          NaN
        </WithStyles(WithStyles(ForwardRef(TableCell)))>
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          className="makeStyles-row-2"
          component="th"
          scope="row"
        >
          Name of the file
        </WithStyles(WithStyles(ForwardRef(TableCell)))>
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          align="right"
        >
          <WithStyles(ForwardRef(Fab))
            aria-label="Delete"
            className="makeStyles-fab-4"
            data-test="button-delete"
            size="small"
          >
            <DeleteOutlinedIcon />
          </WithStyles(ForwardRef(Fab))>
        </WithStyles(WithStyles(ForwardRef(TableCell)))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
</div>
`;
